it/uk/gov/hmrc/agentsubscriptionfrontend/controllers/BusinessIdentificationControllerISpec.scala

```diff
@@ -544,6 +595,148 @@ trait BusinessIdentificationControllerISpec extends BaseISpec with SessionDataMi
+      val doc = Jsoup.parse(bodyOf(result))
+      doc.getElementById("name").`val` shouldBe initialDetails.name
+
+
+      val backLink = doc.getElementsByClass("link-back")
+      backLink.attr("href") shouldBe routes.SubscriptionController.showCheckAnswers().url
+
+      val form = doc.select("form").first()
+      form.attr("method") shouldBe "POST"
+      form.attr("action") shouldBe routes.BusinessIdentificationController.submitBusinessNameForm().url
+    }
+
+    "display business name form if the name is not des complaint" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails.copy(name = "My Agency &"))
+
+      val result = await(controller.showBusinessNameForm(request))
+      result should containMessages("businessName.updated.title", "businessName.updated.p1",  "businessName.description", "businessName.continue.button")
+      val doc = Jsoup.parse(bodyOf(result))
+      doc.getElementById("name").`val` shouldBe "My Agency &"
+
+      val backLink = doc.getElementsByClass("link-back")
+      backLink.attr("href") shouldBe routes.BusinessIdentificationController.showConfirmBusinessForm().url
+
+      val form = doc.select("form").first()
+      form.attr("method") shouldBe "POST"
+      form.attr("action") shouldBe routes.BusinessIdentificationController.submitBusinessNameForm().url
+    }
+
+    "redirect to the /business-type page if there is no InitialDetails in session because the user has returned to a bookmark" in {
+      implicit val request = authenticatedAs(subscribingCleanAgentWithoutEnrolments)
+
+      val result = await(controller.showBusinessNameForm(request))
+
+      resultShouldBeSessionDataMissing(result)
+    }
+  }
+
+  "submitBusinessNameForm" should {
+    behave like anAgentAffinityGroupOnlyEndpoint(request => controller.submitBusinessNameForm(request))
+
+    "update business name after submission" in {
+      implicit val request =
+        authenticatedAs(subscribingCleanAgentWithoutEnrolments).withFormUrlEncodedBody("name" -> "new Agent name")
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails)
+
+      val result = await(controller.submitBusinessNameForm(request))
+      status(result) shouldBe 303
+      redirectLocation(result).head shouldBe routes.SubscriptionController.showCheckAnswers().url
+
+      await(sessionStoreService.fetchInitialDetails).get.name shouldBe "new Agent name"
+    }
+
+    "show validation error when the form is submitted with empty name" in {
+      implicit val request =
+        authenticatedAs(subscribingCleanAgentWithoutEnrolments).withFormUrlEncodedBody("name" -> "")
+      sessionStoreService.currentSession.initialDetails = Some(initialDetails)
+
+      val result = await(controller.submitBusinessNameForm(request))
+
+      result should containMessages("businessName.title", "error.business-name.empty")
```

Comment:
Do we need another test here for the pre-check-answers version of this page, e.g. a test like this one but with:
`result should containMessages("businessName.updated.title", "error.business-name.empty")`